### DISCUSSION TITLE
Opinion text columns -- why so many? which is authoritative?

### DISCUSSION THREAD

**Discussion by rbpasker** [freelawproject/courtlistener#4104] [General]
I'd like to understand how the text columns are used in the opinions table

this image shows2  lines of the table, and the <code>plain_text</code> column and  
other XML and HTML columns

how are these columns all meant to be used? which one is authoritative? 

<img width="1091" alt="PastedGraphic-2" src="https://github.com/freelawproject/courtlistener/assets/87866/723eca6f-740a-4e26-ab3c-ae790d2cb59f">
<img width="1041" alt="PastedGraphic-3" src="https://github.com/freelawproject/courtlistener/assets/87866/8033df35-2e42-46ea-88b8-bcf755a9707a">

**Comment by flooie** [comment]
Hi @rbpasker,

Apologies if some of this is repetitive, but let me start with our data model. If you're not familiar with our data model for Opinions, I recommend reviewing [our API documentation](https://www.courtlistener.com/help/api/rest/) to get acquainted.

Our data model hierarchy for Case Law is structured as follows: Docket → OpinionCluster → Opinion(s). Each OpinionCluster is a collection of Opinions, and each row in the data you posted represents an Opinion. An Opinion captures the individual writings of a judge or a unanimous per curiam decision.

For example, a split 5-4 Supreme Court decision could have two rows: one for the Majority and one for the Dissent, while a 9-0 ruling typically has just one row. This concept often trips up new users of our data, so I wanted to highlight that from the start.

Regarding your question about which source is authoritative, the answer is that it's up to you. Not every source contains the same opinions. Some sources may have extensive state trial court opinions, others may include colonial-era decisions, and yet others may have the latest Supreme Court decisions as they're posted to the Supreme Court website.

We decided early on to keep all the decision information we have to avoid overwriting information or creating complicated mergers.  As well as the fact that some data sources identify each opinion individually with authorship etc.  This seems confusing on its face but allows for finer querying for dissents or a specific judge.

If you're looking for what we expect, I suggest using the following hierarchy, which we mostly use to display content on the site.

1. sub_opinion.html_with_citations
2. sub_opinion.xml_harvard
3. sub_opinion.html_columbia
4. sub_opinion.html_lawbox
5. sub_opinion.html_anon_2020
6. sub_opinion.html
7. sub_opinion.plain_text

Thanks for the question!

**Comment by rbpasker** [comment]
thanks @flooie -- 

I've been through all of the docs and the data model. The data model, however, doesn't have the semantics. 

So my question was not about which **opinion** is authoritative for the case, but rather which **column** is authoritative for the opinion -- sorry for the ambiguousness of the question. 

Do I understand correctly that the 7 textual columns represent 7 different sources? will there never be an 8th (or will you add another column)?

**Comment by flooie** [comment]
@rbpasker

There is no authoritative column as of now.  We will almost certainly add more columns from different sources.  HTML with Citations is somewhat meant to be a __the__ column but as of now it is not.

**Reply by rbpasker** [reply]
the HTML is statically generated and stored in the database? 

or did it come from the source as HTML?
